cmd/ctl/cri/migrate_cri.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cri
+
+import (
+	"errors"
+
+	"github.com/kubesphere/kubekey/cmd/ctl/options"
+	"github.com/kubesphere/kubekey/cmd/ctl/util"
+	"github.com/kubesphere/kubekey/pkg/common"
+	"github.com/kubesphere/kubekey/pkg/pipelines"
+	"github.com/spf13/cobra"
+)
+
+type MigrateCriOptions struct {
+	CommonOptions    *options.CommonOptions
+	ClusterCfgFile   string
+	Kubernetes       string
+	EnableKubeSphere bool
+	KubeSphere       string
+	DownloadCmd      string
+	Artifact         string
+	Type             string
+	Role             string
+}
+
+func NewMigrateCriOptions() *MigrateCriOptions {
+	return &MigrateCriOptions{
+		CommonOptions: options.NewCommonOptions(),
+	}
+}
+
+// NewCmdDeleteCluster creates a new delete cluster command
+func NewCmdMigrateCri() *cobra.Command {
+	o := NewMigrateCriOptions()
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Migrate a container",
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(o.Validate())
+			util.CheckErr(o.Run())
+		},
+	}
+
+	o.CommonOptions.AddCommonFlag(cmd)
+	o.AddFlags(cmd)
+	return cmd
+}
+
+func (o *MigrateCriOptions) Run() error {
+	arg := common.Argument{
+		FilePath:          o.ClusterCfgFile,
+		Debug:             o.CommonOptions.Verbose,
+		KubernetesVersion: o.Kubernetes,
+		Type:              o.Type,
+		Role:              o.Role,
+	}
+	return pipelines.MigrateCri(arg, o.DownloadCmd)
+}
+
+func (o *MigrateCriOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&o.Role, "role", "", "worker", "Role to migrate")
+	cmd.Flags().StringVarP(&o.Type, "type", "", "docker", "Type to migrate")
```

Comment:
It is recommended not to set the default value. And the description needs more detailed, for example, which roles can be specified, and which types can be specified.